Bio/Align/fasta_m8.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2021 by Michiel de Hoon.  All rights reserved.
+#
+# This file is part of the Biopython distribution and governed by your
+# choice of the "Biopython License Agreement" or the "BSD 3-Clause License".
+# Please see the LICENSE file that should have been included as part of this
+# package.
+"""Bio.Align support for output from William Pearson's FASTA alignment tools.
+
+This module contains a parser for output from the FASTA programs generated with
+the '-m 8CB' or '-m 8CC' output formats.
+"""
+import re
+import enum
+import numpy
+from Bio.Align import Alignment
+from Bio.Align import interfaces
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+
+class State(enum.Enum):
+    """Enumerate alignment states needed when parsing a BTOP string."""
+
+    MATCH = enum.auto()
+    QUERY_GAP = enum.auto()
+    TARGET_GAP = enum.auto()
+    NONE = enum.auto()
+
+
+class AlignmentIterator(interfaces.AlignmentIterator):
+    """FASTA output alignment iterator.
+
+    For reading the (pairwise) alignments from the FASTA alignment programs
+    using the '-m 8CB' or '-m 8CC' output formats.
+    """
+
+    def __init__(self, source):
+        """Create an AlignmentIterator object.
+
+        Arguments:
+         - source   - input data or file name
+
+        """
+        super().__init__(source, mode="t", fmt="FASTA")
+        stream = self.stream
+        try:
+            line = next(stream)
+        except StopIteration:
+            raise ValueError("Empty file.") from None
```

Comment:
Not tested - I think I put in a blanket testing of all the SeqIO parsers with an empty file, as part of checking for consistent error messages. I don't know if we're at that point yet with the new alignment suite (no top level API)?

Reply:
I have added a test.